src/Components/Modals/Modal_SignIn/Modal_SignIn.js

```diff
@@ -3,32 +3,82 @@ import "./Modal_SignIn.css";
 import { connect } from 'react-redux';
 import OutsideClick from '../../../OutsideClick';
 import Actions from "../../../Actions/actions";
-import Utils from '../../../utils'
+import Utils from '../../../utils';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faExclamationCircle } from '@fortawesome/free-solid-svg-icons';
+library.add(faExclamationCircle)
 
 class ModalSignIn extends Component {
+    constructor() {
+        super();
+        this.state = {
+            username: "",
+            password: "",
+        }
+    }
+
+    onSignInClicked() {
+        this.props.onSignInClicked(this.state);
+    }
+
+    handleSignUpClick() {
+        this.props.toggleModal('signIn', false);
+        this.props.toggleModal('signUp', true);
```

Comment:
It feels weird that you have to emit 2 actions to do this. 2 modals can't be open at the same time right? So the logic for closing a modal should be implicit in the action for opening one, ie you should only have to emit one action.

You can tackle this in a small followup PR.

Reply:
sounds good; i did also notice that problem but didn't fix it cuz it will be a bit of work. I will do it after closing this PR

Reply:
ok